commit 5f3bceb68dd211be977eb61d5f1ea68e7de51b7a
Author: Ulrich Weigand <ulrich.weigand@de.ibm.com>
Date:   Tue Sep 6 17:22:51 2016 +0200

    Fix typo in ada_language_arch_info
    
    This fixes a bug introduced by a wrong replacement here:
    https://sourceware.org/ml/gdb-patches/2007-06/msg00196.html
    
    The Ada "long_long_float" type is supposed to correspond to the
    platform ABI long double type, not double.
    
    gdb/ChangeLog:
    
            * ada-lang.c (ada_language_arch_info): Use gdbarch_long_double_bit
            instead of gdbarch_double_bit for "long_long_float".
    
    Signed-off-by: Ulrich Weigand <ulrich.weigand@de.ibm.com>

### a/gdb/ChangeLog
### b/gdb/ChangeLog
## -1,3 +1,8 @@
+2016-09-05  Ulrich Weigand  <uweigand@de.ibm.com>
+
+	* ada-lang.c (ada_language_arch_info): Use gdbarch_long_double_bit
+	instead of gdbarch_double_bit for "long_long_float".
+
 2016-09-05  Pedro Alves  <palves@redhat.com>
 
 	* NEWS: Mention that a C++ compiler is now required.
--- a/gdb/ada-lang.c
+++ b/gdb/ada-lang.c
@@ -14012,7 +14012,7 @@ ada_language_arch_info (struct gdbarch *gdbarch,
     = arch_integer_type (gdbarch, gdbarch_long_long_bit (gdbarch),
 			 0, "long_long_integer");
   lai->primitive_type_vector [ada_primitive_type_long_double]
-    = arch_float_type (gdbarch, gdbarch_double_bit (gdbarch),
+    = arch_float_type (gdbarch, gdbarch_long_double_bit (gdbarch),
 		       "long_long_float", NULL);
   lai->primitive_type_vector [ada_primitive_type_natural]
     = arch_integer_type (gdbarch, gdbarch_int_bit (gdbarch),
